### ISSUE TITLE
Annotators are executed in wrong buffer for recursive minibuffers

### ISSUE THREAD

**Issue by minad** [minad/marginalia#111]
https://github.com/minad/marginalia/blob/86ac625169041cdc706c5e39cae0bf314c042473/marginalia.el#L975-L979

For recursive minibuffers one would expect to see the variable value annotators for the parent minibuffer if C-hv was started from the parent minibuffer. If a recursive C-hv was started from another window one would expect the variable values from that window. In any case the annotators should run in the window/buffer which was selected right before starting the recursive minibuffer. cc @oantolin

**Comment by minad** [comment]
One could advise completing-read and store the current buffer. Emacs 28 has minibuffer--original-buffer for this purpose.

**Comment by minad** [comment]
Close this as wontfix, since I don't see a good and simple approach to fix this. It is a minor issue anyway.
